### ISSUE TITLE
tracking: Stop batch tasks more timely.

### ISSUE THREAD

**Issue by liurenjie1024** [risingwavelabs/risingwave#9469]
For details please refere to #9418.

- [x] HashJoinExecutor
- [ ] NestedLoopJoinExecutor
- [ ] LocalLookupJoinExecutor
- [ ] DistributedLookupJoinExecutor
- [ ] ExpandExecutor
- [ ] FilterExecutor
- [ ] GenericExchangeExecutor
- [ ] GroupTopNExecutor
- [ ] HashAggExecutor
- [ ] HopWindowExecutor
- [ ] LimitExecutor
- [ ] MergeSortExchangeExecutor
- [ ] SortExecutor
- [ ] ProjectExecutor
- [ ] ProjectSetExecutor
- [ ] RowSeqScanExecutor
- [ ] SortAggExecutor
- [ ] SourceExecutor
- [ ] SysRowSeqScanExecutor
- [ ] TableFunctionExecutor
- [ ] TopNExecutor
- [ ] UnionExecutor
- [ ] ValuesExecutor

**Comment by lmatz** [comment]
Let's also merge into v0.19.0-rc when this issue is completely done.

**Comment by liurenjie1024** [comment]
> Let's also merge into v0.19.0-rc when this issue is completely done.

Yes, @ZENOTME  has found a good solution so that we don't need to modify executors one by one.

**Comment by lmatz** [comment]
Stop in the wrapper executor?

**Comment by liurenjie1024** [comment]
> Stop in the wrapper executor?

You're so successful.

**Comment by lmatz** [comment]
Please also make a PR including all batch-related fixes to v0.19.0-rc when there is capacity, thanks!

**Comment by liurenjie1024** [comment]
cc @BugenZhao
